src/screens/Registration/Registration.js

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { Text, ScrollView, KeyboardAvoidingView } from "react-native";
+import { styles } from "./styles";
+import { Button, Input } from "../../components";
+
+const fields = {
+  email: "email",
+  password: "password",
+  firstName: "firstName",
+  lastName: "lastName",
+  phone: "phone"
+};
+
+const Registration = props => {
+  const {
+    emailDefault,
+    passwordDefault,
+    firstNameDefault,
+    lastNameDefault,
+    phoneDefault,
+    hintText
+  } = props;
+  const [
+    { email, password, firstName, lastName, phone },
+    setRegistrationData
+  ] = useState({
+    [fields.email]: emailDefault,
+    [fields.password]: passwordDefault,
+    [fields.firstName]: firstNameDefault,
+    [fields.lastName]: lastNameDefault,
+    [fields.phone]: phoneDefault
+  });
+
+  const setRegistrationDataCommon = field => value =>
+    setRegistrationData(prevData => ({ ...prevData, [field]: value }));
+
+  const onRegisterPress = () => {
+    props.onSubmit();
+  };
+  const onBackPress = () => {
+    // TODO: figure out whether it's possible just to trigger default android back button
+    props.onBackNavigation();
+  };
+
+  return (
+    <ScrollView
+      contentContainerStyle={styles.scrollContainer}
+      showsVerticalScrollIndicator={false}
+      enabled
+    >
+      <KeyboardAvoidingView style={styles.container} enabled>
+        <Text style={styles.headerText}>Регистрация</Text>
+        <Input
+          value={email}
+          label="E-mail"
+          textContentType="emailAddress"
+          onChangeText={setRegistrationDataCommon(fields.email)}
+        />
+        <Input
+          value={password}
+          label="Пароль"
+          textContentType="password"
+          onChangeText={setRegistrationDataCommon(fields.password)}
+        />
+        <Text style={styles.hintText}>{hintText}</Text>
+        <Input
+          value={firstName}
+          label="Имя"
+          textContentType="name"
+          onChangeText={setRegistrationDataCommon(fields.firstName)}
+        />
+        <Input
+          value={lastName}
+          label="Фамилия"
+          textContentType="familyName"
+          onChangeText={setRegistrationDataCommon(fields.lastName)}
+        />
+        <Input
+          value={phone}
+          label="Телефон"
+          textContentType="telephoneNumber"
+          onChangeText={setRegistrationDataCommon(fields.phone)}
+        />
+        <Button
+          caption="Регистрация"
+          onPress={onRegisterPress}
+          appearance="Dark"
+        />
+        <Button caption="НАЗАД" onPress={onBackPress} appearance="Borderless" />
```

Comment:
We need to use localization here.

Reply:
At the moment we don't have dictionary with translations, so I guess it can be modified further

Reply:
I don't think so. It would be hard to find all places with text. Let's start now. We can support one language for now.